### DISCUSSION TITLE
html, body css not working in pages (Need Help)

### DISCUSSION THREAD

**Discussion by tajsoheil786** [vercel/next.js#66399] [Help] [answered]
### Summary

I have the below CSS code in globals.css which is working perfectly in app pages but I cannot set these in pages, any way to do that? I am using v14 of nextjs :

html, body {
  margin: 0px !important;
  padding: 0px;
  box-sizing: border-box;
}

**Comment by icyJoseph** [comment]
Are you importing this file into the `_app` file?

**Reply by tajsoheil786** [reply]
There is no _app file in nextjs v 14.1.4 😅?

**Reply by icyJoseph** [reply]
but you are saying:

> app pages but I cannot set these in pages

Could you clarify that? it makes one think that you are migrating `pages router` to `app router` or keeping both.

Also, Next v14 supports both App Router and Pages Router. Let's not make the distinction between those by using the Next.js version, as that are grounds for confusion.

**Reply by sarfarazsiddiquii** [reply]
Actually the css file (global.css) in the `app` directory has the code

html, body {
margin: 0px !important;
padding: 0px;
box-sizing: border-box;
}

is not being applied to any of the pages.
and if I try to use the above code in my filename.module.css it says you cannot use `body` selector.

**Reply by sarfarazsiddiquii** [reply]
i am struggling with the 8px margin over the header and footer in my pages.

**Reply by icyJoseph** [reply]
Right, but are you importing it as `import 'global.css'` or something like that?

You need to import it at the `layout.tsx`, at the root, so called, root layout.

**Reply by sarfarazsiddiquii** [reply]
im importing global.css the the top of layout.tsx as 

`import "./global.css";`

**Reply by icyJoseph** [reply]
Do you have a link to a repository to take look?

**Comment by sarfarazsiddiquii** [comment]
Unfortunately, the repo is private and I can't share it due to privacy reasons, 
I'll try to explain the problem.

I'm working with nextjs v14.1.4 and I've created multiple pages eg - about page, everything works fine for about.tsx there is css for for that page named about.module.css however I'm dealing with the problem of 8px margin along the outer parts of the screen.
I tried fixing it by removing any padding or margin in about.module.css,
I tried putting the above code in global.css, 
but the margin stays the same.

**Reply by icyJoseph** [reply]
This is very confusing though, `about.tsx` would only work in Pages Router.

Are you maybe mixing concepts here, because you seem to think that Next 14 is automatically App Router, and that's not the case.

Could you share your file structure at least?

**Reply by sarfarazsiddiquii** [reply]
yes sure,

├─ app
│  ├─ global.css
│  ├─ layout.tsx
│  ├─ page.module.css
│  └─ page.tsx
├─ next.config.mjs
├─ package-lock.json
├─ package.json
├─ pages
│  ├─ about.module.css
│  ├─ about.tsx
│  ├─ number
│  │  ├─ [number].module.css
│  │  └─ [number].tsx
│  ├─ privacy-policy.module.css
│  ├─ privacy-policy.tsx
│  ├─ range
│  │  ├─ [range].module.css
│  │  └─ [range].tsx
│  ├─ sitemap
│  │  ├─ sitemap.module.css
│  │  └─ [sitemap].tsx
│  ├─ sitemap.module.css
│  ├─ sitemap.tsx
│  └─ _document.js
├─ postcss.config.js
├─ public
├─ README.md
├─ tailwind.config.ts
└─ tsconfig.json

**Reply by icyJoseph** [reply, marked as answer]
Right, so you do need an `_app`.

Pages Route and App Router, are fundamentally different. They behave as two different applications, co-existing under the same repository. 

You need to create a standard `_app` file, and do this:

```tsx
import '../app/global.css'; // do the right path here, I am on a bit of a hurry now
import type { AppProps } from 'next/app';
 
export default function MyApp({ Component, pageProps }: AppProps) {
  return <Component {...pageProps} />;
}
```

**Reply by tajsoheil786** [reply]
Thanks a lot, Joseph! It worked!

**Comment by sarfarazsiddiquii** [comment]
Thanks! it worked.
